libraries/chain/database_worker_objects.cpp

```diff
@@ -159,14 +159,11 @@ namespace golos { namespace chain {
 
         asset consumption;
 
-        auto month_end = now + fc::days(30);
-        for (auto wto_itr = wto_idx.lower_bound(now); wto_itr != wto_idx.end() && wto_itr->next_cashout_time <= month_end; ++wto_itr) {
-            auto payments = 1;
-            payments += ((month_end - wto_itr->next_cashout_time).to_seconds() / wto_itr->payments_interval);
-
+        auto month_sec = fc::days(30).to_seconds();
+        for (auto wto_itr = wto_idx.begin(); wto_itr != wto_idx.end() && wto_itr->next_cashout_time < time_point_sec::maximum(); ++wto_itr) {
             auto payment = (wto_itr->specification_cost + wto_itr->development_cost) / wto_itr->payments_count;
 
-            consumption += (payment * payments);
+            consumption += (payment * (month_sec / wto_itr->payments_interval));
```

Comment:
1. what if there is only one payment left for current work, and it's `payments_interval` set to 1 day?
2. last payment can be slightly larger then others due truncation when divide (e.g. 1.000 GOLOS and 3 intervals)

Reply:
@zxcat 1 resolved

Reply:
@zxcat 2 resolved https://github.com/GolosChain/golos/pull/1092/commits/f01af4bb1f2cd49eb9d4037c4151d9c92843ea61